traffic_ops/traffic_ops_golang/deliveryservice/validate.go

```diff
@@ -0,0 +1,365 @@
+func ValidMidHeaderRewrite(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidMissLat ...
+func ValidMissLat(ds tc.DeliveryService) error {
+	if math.Abs(ds.MissLat) > 90 {
+		return fmt.Errorf("missLat value %2.0f must not exceed +/- 90.0", ds.MissLat)
+	}
+	return nil
+}
+
+// ValidMissLong ...
+func ValidMissLong(ds tc.DeliveryService) error {
+	if math.Abs(ds.MissLong) > 90 {
+		return fmt.Errorf("missLong value %2.0f must not exceed +/- 90.0", ds.MissLong)
+	}
+	return nil
+}
+
+// ValidMultiSiteOrigin ...
+func ValidMultiSiteOrigin(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidMultiSiteOriginAlgorithm ...
+func ValidMultiSiteOriginAlgorithm(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidOrgServerFqdn ...
+func ValidOrgServerFqdn(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidOriginShield ...
+func ValidOriginShield(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidProfileID ...
+func ValidProfileID(ds tc.DeliveryService) error {
+	// TODO: validate exists in profile table
+	return nil
+}
+
+// ValidProtocol ...
+func ValidProtocol(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidQstringIgnore ...
+func ValidQstringIgnore(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidRangeRequestHandling ...
+func ValidRangeRequestHandling(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidRegexRemap ...
+func ValidRegexRemap(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidRegionalGeoBlocking ...
+func ValidRegionalGeoBlocking(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidRemapText ...
+func ValidRemapText(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidRoutingName ...
+func ValidRoutingName(ds tc.DeliveryService) error {
+	if len(ds.RoutingName) > 48 {
+		return fmt.Errorf("routing name '%s' can be no longer than 48 characters", ds.RoutingName)
+	}
+	return nil
+}
+
+// ValidSigningAlgorithm ...
+func ValidSigningAlgorithm(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidSslKeyVersion ...
+func ValidSslKeyVersion(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidTenantID ...
+func ValidTenantID(ds tc.DeliveryService) error {
+	// TODO: validate exists in tenant table
+	return nil
+}
+
+// ValidTrRequestHeaders ...
+func ValidTrRequestHeaders(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidTrResponseHeaders ...
+func ValidTrResponseHeaders(ds tc.DeliveryService) error {
+	return nil
+}
+
+// ValidTypeID ...
+func ValidTypeID(ds tc.DeliveryService) error {
+	// TODO: validate exists in type table
+	return nil
+}
+
+// ValidXMLID ...
+func ValidXMLID(ds tc.DeliveryService) error {
+	if len(ds.XMLID) == 0 {
+		return fmt.Errorf("xmlId is required")
+	}
+	if strings.IndexFunc(ds.XMLID, unicode.IsSpace) != -1 {
+		return fmt.Errorf("xmlId '%s' must not contain spaces", ds.XMLID)
+	}
+	if len(ds.XMLID) > 48 {
+		return fmt.Errorf("xmlId '%s' must be no longer than 48 characters", ds.XMLID)
+	}
+	return nil
+}
```

Comment:
This file is mostly a stub. I'd still consider replacing the `// Name ...` comments with `TODO: Basic description of validation` comments. It'll help us remember to fill these out and also help us remember what each thing is supposed to be checking for. That might be more than you've got time for, though, so this is mostly fine as a stub for now.